### ISSUE TITLE
Theme song ending in the middle of the game.

### ISSUE THREAD

**Issue by JacobChrist** [spherous/ArtCenterJam2022#1]
Strangely, when the music ends just before midnight it is actually the end of the track (4:54)  So the 300 second timer is running slow.

I tracked this down to the calculation of ellapsedDuration in the DayNightCycle class here:
```
        ellapsedDuration += Time.deltaTime;
```
And changed it to this:
```
        ellapsedDuration = Time.timeSinceLevelLoad;
```

Accumulating time using Time.deltaTime introduces an integration error which in the case of game is very significant (more than double).  I pushed this change.

So when we showed the game yesterday because the we didn't start the game right at the beginning of the presentation and because the song ended during the demo, and because we talked with Christophe beyond the end of the game we got more than our fair share of five minutes.

So now the day night cycle indicator now get to midnight at 5:00 (300 seconds) and the game ends.  I need to get to work so I do not have time to chase the problem down to completion.

**Comment by JacobChrist** [comment]
Adding these notes here for myself because i want to read them later:

https://blog.unity.com/technology/fixing-time-deltatime-in-unity-2020-2-for-smoother-gameplay-what-did-it-take

https://answers.unity.com/questions/612133/accumulating-deltatime-oddity.html

**Comment by JacobChrist** [comment]
@spherous Had some more time to absorbed your code today.  I think I understand what you were saying now.  You are dividing up time into cycles (in this case two).  Each cycleDuration is 1/numberOfCycles time through the level.  By not using  Time.timeSinceLevelLoad the time percentage of the cycle cannot be reset.

So based on all this the real problem is that the cycleDuration is set to 300 second (or five minutes).  Since there are two cycles the total time from dusk to dawn is 10 minutes.  Since the theme is 4:56 (296 seconds) the cycle duration should be 296/numberOfCycles = 148.

I created a new var to keep track of the start of a cycle and changed the elapsedCycleDuration to this:
```
        elapsedCycleDuration = Time.time - cycleStartTime; //+= Time.deltaTime;
```

This prevents the float integration error but restores the delta time calculation from the start of the cycle.  If you want to pause you just need to subtract out the pause time now.

I've also changed some names of variable in the DayNightCycle.cs file to clarify what some of the magic numbers are for my old feeble brain.

I didn't update the cycleDuration on AvaeScene yet, only JacobScene.

**Comment by JacobChrist** [comment]
If we try to implement Stimac's ending, the cycle duration need to be 126 seconds.

**Comment by JacobChrist** [comment]
There is another issue.  I saw it earlier but shrugged it off because it was working in the Unity Editor.  Its NOT reliable to do an equality comparison of floating point values.  It is advisable to use >= or <= when comparing two floats.  In the WebGL player the exact moment when the first cycle ended was not being captured and the cycle never flipped to becomingDay.  I have corrected this in the latest commit  https://github.com/spherous/ArtCenterJam2022/commit/9b45f7ef531f362bae435175bf455a2ccbb0d4e4 .
